### PR TITLE
MODSOURCE-216: Update MARC 005 field when MARC record has changes

### PR DESCRIPTION
**Pull request by Aliaksandr-Fedasiuk**

### CODE REVIEW

File: mod-source-record-storage-server/src/main/java/org/folio/services/handlers/InstancePostProcessingEventHandler.java [hunk 1]
```diff
@@ -17,24 +18,31 @@
 import org.folio.dao.util.ParsedRecordDaoUtil;
 import org.folio.processing.events.services.handler.EventHandler;
 import org.folio.processing.exceptions.EventProcessingException;
-import org.folio.rest.jaxrs.model.AdditionalInfo;
-import org.folio.rest.jaxrs.model.EntityType;
-import org.folio.rest.jaxrs.model.ExternalIdsHolder;
-import org.folio.rest.jaxrs.model.Record;
+import org.folio.processing.mapping.defaultmapper.processor.parameters.MappingParameters;
+import org.folio.rest.jaxrs.model.*;
```
Comment (OleksiiKuzminov) [inline mod-source-record-storage-server/src/main/java/org/folio/services/handlers/InstancePostProcessingEventHandler.java:22]: Dont use wildcards

File: mod-source-record-storage-server/src/main/java/org/folio/services/handlers/InstancePostProcessingEventHandler.java [hunk 2]
```diff
@@ -175,6 +186,47 @@ private void setInstanceIdToRecord(Record record, JsonObject instance) {
...
+    if (isField005NeedToUpdate(record, context)) {
+      String date = AdditionalFieldsUtil.dateTime005Formatter.format(ZonedDateTime.ofInstant(Instant.now(), ZoneId.systemDefault()));
+      boolean isLatestTransactionDateUpdated = AdditionalFieldsUtil.addControlledFieldToMarcRecord(record, AdditionalFieldsUtil.TAG_005, date, true);
+      if (!isLatestTransactionDateUpdated) {
+        throw new PostProcessingException(format("Failed to update field '005' to record with id '%s'", record.getId()));
+      }
+    }
+  }
+
+  private boolean isField005NeedToUpdate(Record record, HashMap<String, String> context) throws JsonProcessingException {
```
Comment (OleksiiKuzminov) [inline mod-source-record-storage-server/src/main/java/org/folio/services/handlers/InstancePostProcessingEventHandler.java:199]: Plz move all field methods to the additional field util

**Review by OleksiiKuzminov — CHANGES_REQUESTED** [review]
Plz update news.md

File: mod-source-record-storage-server/src/main/java/org/folio/services/handlers/InstancePostProcessingEventHandler.java (hunk 1 above)
Comment (Aliaksandr-Fedasiuk) [reply mod-source-record-storage-server/src/main/java/org/folio/services/handlers/InstancePostProcessingEventHandler.java]: Fixed.

File: mod-source-record-storage-server/src/main/java/org/folio/services/handlers/InstancePostProcessingEventHandler.java (hunk 2 above)
Comment (Aliaksandr-Fedasiuk) [reply mod-source-record-storage-server/src/main/java/org/folio/services/handlers/InstancePostProcessingEventHandler.java]: Fixed.

**Comment by Aliaksandr-Fedasiuk** [comment]
> Plz update news.md
Ready.

**Review by OleksiiKuzminov — APPROVED** [review]

**Review by RuslanLavrov — APPROVED** [review]

**Review by VRohach — APPROVED** [review]

**Review by KaterynaSenchenko — APPROVED** [review]

**Comment by Aliaksandr-Fedasiuk** [comment]
Closed. Reason: some commits made with incorrect email (can't sign CLA).

**Pull request closed without merging**